### ISSUE TITLE
[Bug]: [string_match] The result for search/query is empty when using string matching operator "_"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#30721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240218-99297ab8
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0rc35
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The result for string matching operator "_" is empty

 ```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "string like '_0'", output_fields=['string'])
>>>
>>> res
['[]']
```

### Expected Behavior

The string "10", "20"... should be matched

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility
connections.connect()

dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, string_field, float_vector])

import numpy as np
collection = Collection("test_search_collection_binbin_tmp_0", schema=schema)
nb = 2000
import random
default_search_params = {"metric_type": "L2", "params": {"nprobe": 100}}
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [str(i) for i in range(nb)], vectors])
index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
collection.create_index("float_vector", index_param, index_name="a")
collection.load()
nq = 1
limit = 1000
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "string like '_0'", output_fields=['string'])
res[0]
collection.query("string like '_0')
```

**Comment by longjiquan** [comment]
already fixed by #30826

**Comment by binbinlv** [comment]
Verified and fixed:
milvus: master-20240229-dcfc3531-amd64
pymilvus: 2.4.0rc37
```
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "string like '_0'", output_fields=['string'])
>>>
>>> len(res[0])
9
>>> res[0]
["id: 20, distance: 19.193359375, entity: {'string': '20'}", "id: 60, distance: 20.480823516845703, entity: {'string': '60'}", "id: 40, distance: 21.84343910217285, entity: {'string': '40'}", "id: 50, distance: 22.566959381103516, entity: {'string': '50'}", "id: 80, distance: 22.884328842163086, entity: {'string': '80'}", "id: 70, distance: 23.627620697021484, entity: {'string': '70'}", "id: 90, distance: 24.3341121673584, entity: {'string': '90'}", "id: 10, distance: 25.377979278564453, entity: {'string': '10'}", "id: 30, distance: 27.238801956176758, entity: {'string': '30'}"]
>>>
>>> collection.query("string like '_0'")
[{'int64': 10}, {'int64': 20}, {'int64': 30}, {'int64': 40}, {'int64': 50}, {'int64': 60}, {'int64': 70}, {'int64': 80}, {'int64': 90}]
```
